sstables/sstable_set.cc

```diff
@@ -852,6 +857,14 @@ class incremental_reader_selector : public reader_selector {
 
         return {};
     }
+
+    // Can be false-positive but never false-negative!
+    virtual bool has_new_readers(const std::optional<dht::ring_position_view>& pos) const noexcept override {
+#ifndef SCYLLA_BUILD_MODE_RELEASE
+        assert(!pos || dht::ring_position_tri_compare(*_s, *pos, dht::ring_position_view::for_range_end(*_pr)) <= 0);
+#endif
+        return !end_of_stream() && (!pos || dht::ring_position_tri_compare(*_s, *pos, _selector_position) >= 0);
```

Comment:
Looks like the logic is the same here as in the base class, the only difference is what pos we consider as `max`. S maybe we just need a `virtual bool is_max_pos()` and keep `has_new_readers()` non-virtual.